File: Minsk/CodeAnalysis/Syntax/SyntaxToken.cs
using System.Text;

namespace Minsk.CodeAnalysis.Syntax;

public sealed class SyntaxToken : SyntaxNode
{
    public override SyntaxKind Kind { get; }
    public override IEnumerable<SyntaxNode> Children => Enumerable.Empty<SyntaxNode>();
    public string Text { get; }
    public int Position { get; }
    public object? Value { get; }

    public SyntaxToken(SyntaxKind kind, string text, int position, object? value)
    {
        Kind = kind;
        Text = text;
        Position = position;
        Value = value;
    }

    public override string ToString()
    {
        var result = new StringBuilder();
        result.Append($"{Kind} '{Text}'");
        if (Value != null)
        {
            result.Append($" {Value}");
        }

        return result.ToString();
    }
}

File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/syntax/SyntaxToken.kt
package dev.phytolizer.minsk.analysis.syntax

class SyntaxToken(override val kind: SyntaxKind, val position: Int, val text: String, val value: Any?) : SyntaxNode() {
    override fun toString(): String {
        val result = StringBuilder()
        result.append("$kind '$text'")
        if (value != null) {
            result.append(" $value")
        }
        return result.toString()
    }

    override val children: List<SyntaxNode>
        get() = listOf()
}
